configure.ac

```diff
@@ -22,6 +22,10 @@ PKG_CHECK_MODULES([TSS2_SYS], [tss2-sys >= 2.0.0])
 PKG_CHECK_MODULES([CRYPTO], [libcrypto >= 1.0.2g])
 PKG_CHECK_MODULES([CURL], [libcurl])
 
+PKG_CHECK_VAR(bashcompdir, [bash-completion], [completionsdir], ,
+  bashcompdir="${sysconfdir}/bash_completion.d")
```

Comment:
On Fedora this sets `bashcompdir` to `/usr/share/bash-completion/completions/`. I also thought that was the correct path but when testing your patch it didn't work.

By looking at the code in `/usr/share/bash-completion/bash_completion`, it seems that `__load_completion()` is called lazily when completion has to be made for a command line tool. Now, this searches for a completion script in `/usr/share/bash-completion/completions/$cmd` or `/usr/share/bash-completion/completions/$cmd.bash`.

For example for `tpm2_create` it will try to source a `/usr/share/bash-completion/completions/tpm2_create` or `/usr/share/bash-completion/completions/tpm2_create.bash`.

So it seems that its designed to have a completion that matches the command name. This would work if we had subcommands (i.e: tpm2 create, tpm2 load, etc) but it won't work in our case since every command is named differently.

Instead, for "global" completion that covers different commands, the `/etc/bash_completion.d` directory should be used instead. I see that the files there are sourced unconditionally.

So I think your `configure.ac` snippet should instead be:

```
PKG_CHECK_VAR(bashcompdir, [bash-completion], [compatdir], ,
  bashcompdir="${sysconfdir}/bash_completion.d")
AC_SUBST(bashcompdir)
```
In other words, using `compatdir` instead of `completionsdir` by looking at `/usr/share/pkgconfig/bash-completion.pc`:

```
prefix=/usr
compatdir=/etc/bash_completion.d
completionsdir=${prefix}/share/bash-completion/completions
helpersdir=${prefix}/share/bash-completion/helpers

Name: bash-completion
Description: programmable completion for the bash shell
URL: https://github.com/scop/bash-completion
Version: 2.8
```

@kaiserkyu I'm not a bash completion expert though so please let me know if I got something wrong.